index.js

```diff
@@ -9,10 +9,17 @@ from2.obj = obj
 var Proto = ctor()
 
 function from2(opts, read) {
-  if (typeof opts === 'function') {
+  if (typeof opts !== 'object' || Array.isArray(opts)) {
     read = opts
     opts = {}
   }
+  
+  if(Array.isArray(read)) {
+    read = read.slice()
+    return from2(function (_, cb) {
+      cb(null, read.length ? read.shift() : null)
+    })
+  }
```

Comment:
We need to support passing in the options as well (to support objectMode)
Could be implemented like this

``` js
var list = read.slice()
read = function(_, cb) {
  cb(null, list.length ? list.shift() : null)
}
```